### ISSUE TITLE
Prefer-template multiline

### ISSUE THREAD

**Issue by Cellule** [eslint/eslint#3507]
Sometimes we break string in mulitple lines for lisibility like
```js
const a = "that string can get very" +
  `long ${someVar} and we don't want it on one line`
```

The rule warns us to use template string when this is not needing and defeats the purpose.
Right now, there is a work around to add 2 template strings like

```js
const a = `string1` +
  `string2`;
```

But I don't know if this is something we want/should keep. see #3506

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by ilyavolodin** [comment]
@Cellule Could you paste ESLint output for the code that errors?

**Comment by mysticatea** [comment]
Thank you for the report.

I confirmed that.

```js
/* eslint-env es6 */
/* eslint prefer-template:2 */
const a = "that string can get very" +
  `long ${someVar} and we don't want it on one line`;
```

```
> eslint -v
v1.2.1
> eslint test.js --no-eslintrc

test.js
  3:11  error  Unexpected string concatenation  prefer-template

✖ 1 problem (1 error, 0 warnings)
```

I thought this is a false positive.
Two or more string (and template literal) concatenation should be usable to make data without line breaks as keeping readability.

And we should rethink should the next pattern be warned?

```js
let a = "this is a long string" +
    list.map(item => item.name).join(", ") +
    "and more long strings";
```

**Comment by mysticatea** [comment]
Ah, I could write that like:

```js
let a = `this is a long string ${
    list.map(item => item.name).join(", ")
    } and more long strings`;
```

**Comment by IanVS** [comment]
For multi-line concatenation, perhaps the rule should check that the string contains a `\n` at the end?  Because we would still want the rule to flag:

```js
const a = "This is a multi line string.\n" +
    "It continues onto another line"
```

Which can be replaced by:
```js
const a = `This is a multi line string.
    It continues onto another line.`
```

**Comment by mathieumg** [comment]
@IanVS I imagine the whitespace on the second line of the second example is accidental, because the two versions are not currently equivalent.

**Comment by IanVS** [comment]
Yes, you're right.  You can tell I've not used them much yet.  :flushed: 

Should be?

```js
const a = "This is a multi line string.\n" +
    "It continues onto another line"
```

```js
const a = `This is a multi line string.
It continues onto another line.`
```

Hm, that's kinda ugly.  :-/

**Comment by Cellule** [comment]
In my opinion, if the 2 strings are not on the same line, we should not warn to use template string since it is not always preferable for lisibility

**Comment by firedev** [comment]
Hey, what about this
```
        loader: ExtractText.extract('style',
            'css?sourceMap!sass?sourceMap&outputStyle=expanded&' +
            'includePaths[]=' +
            encodeURIComponent(path.resolve(__dirname, './src')) + '&' +
            'includePaths[]=' +
            encodeURIComponent(path.resolve(__dirname,
              './src/vendor/97f4b8a9aef3-with-fixes-for-node-sass/scss')) + '&' +
            'includePaths[]=' +
            encodeURIComponent(path.resolve(__dirname, './node_modules/compass-mixins/lib'))),
      },
```

**Comment by mathieumg** [comment]
It seems @Cellule 's concern in https://github.com/eslint/eslint/issues/3507#issuecomment-134471387 wasn't addressed. I'll go on a bit of self-promotion and link you to https://github.com/mathieumg/tempura#nowhitespace which I use to build long URIs and URLs using template literals.
